plugin/plugin.go

```diff
@@ -11,8 +11,10 @@ import (
 
 type Plugin[T any] struct {
 	State         *T
-	rpcMethod     map[string]*rpcMethod[T]
-	notification  map[string]*rpcNotification[T]
+	rpcMethods     map[string]*rpcMethod[T]
+	notifications  map[string]*rpcNotification[T]
+	hooks          map[string]*rpcHook[T]
+	subscriptions  map[string]*rpcNotification[T]
```

Comment:
```suggestion
	RpcMethods     map[string]*rpcMethod[T]
	Notifications  map[string]*rpcNotification[T]
	Hooks          map[string]*rpcHook[T]
	Subscriptions  map[string]*rpcNotification[T]
```

I think we need to make this public otherwise we can not access to it inside the get manifest.